word_nerd/decks/views.py

```diff
@@ -36,3 +36,22 @@ def get(self, request, language_pk):
         decks = Deck.objects.filter(language=language, owner=request.user)
         serializer = DeckSerializer(decks, many=True)
         return Response({'decks': serializer.data}, status=status.HTTP_200_OK)
+
+
+class EditDeckAPIView(APIView):
+    permission_classes = [IsAuthenticated]
+
+    def post(self, request, pk):
+        try:
+            deck = Deck.objects.get(id=pk, owner=request.user)
+        except ObjectDoesNotExist:
+            return Response(status=status.HTTP_404_NOT_FOUND)
```

Comment:
Would it be better to return `HTTP_400_BAD_REQUEST` here as well? Not sure

Reply:
I feel like it makes more sense to return HTTP_404_NOT_FOUND when the deck/language instance with a specific PK doesn't exist and HTTP_400_BAD_REQUEST when a serializer is invalid :smiley: